R/drive_trash.R

```diff
@@ -0,0 +1,111 @@
+#' Trash file on Google Drive.
+#'
+#' @template file
+#' @template verbose
+#'
+#' @return Logical, indicating whether the trashing succeeded.
+#' @export
+#' @examples
+#' \dontrun{
+#' ## Trash a single file in the trash.
+#' drive_trash("chickwts.csv")
+#'
+#' ## Trash multiple files in trash.
+#' drive_trash(c("abc", "def"))
+#' }
+drive_trash <- function(file = NULL, verbose = TRUE) {
+  trash_file <- as_dribble(file)
+  if (!some_files(trash_file) && verbose) {
+    message(glue("No such files found to trash"))
+  }
+
+  out <- purrr::map_lgl(trash_file$id, trash_one)
+
+  if (verbose) {
+    if (any(out)) {
+      successes <- glue_data(trash_file[out, ], "  * {name}: {id}")
+      message(collapse(c("Files trashed:", successes), sep = "\n"))
+    }
+    if (any(!out)) {
+      failures <- glue_data(trash_file[!out, ], "  * {name}: {id}")
+      message(collapse(c("Files NOT trashed:", failures), sep = "\n"))
+    }
+  }
```

Comment:
Do you think it's possible to fail and get this far when we've failed to trash the file? I guess I could believe it if we were using `purrr::safely()` to wrap `trash_one` and, perhaps, we should be. I suppose you could fail it you don't own the file. Maybe you should explore that scenario to think about how these vectorized functions should work when failure is possible and doesn't raise an error.